lib/slayer/result_matcher.rb

```diff
@@ -0,0 +1,72 @@
+module Slayer
+  class ResultMatcher
+    attr_reader :result, :command
+
+    def initialize(result, command)
+      @result = result
+      @command = command
+
+      @status = result.status || :default
+
+      @handled_default_pass = false
+      @handled_default_fail = false
+
+      # These are set to false if they are never set. If they are set to `nil` that
+      # means the block intentionally passed `nil` as the block to be executed.
+      @matching_block       = false
```

Comment:
They are intended to be private variables, so we don't want a public reader. Nobody outside the internal implementation of this class has the right to know the state of these values.